source/plugins/utils/extension.rb

```diff
@@ -0,0 +1,77 @@
+require "socket"
+require "msgpack"
+require "securerandom"
+require "singleton"
+require_relative "omslog"
+require_relative "constants"
+require_relative "ApplicationInsightsUtility"
+
+
+class Extension

```

Comment:
[pedantic] Extension is not a very descriptive name for this class, I wouldn't know what it is if I saw it for the first time.

Reply:
> [pedantic] Extension is not a very descriptive name for this class, I wouldn't know what it is if I saw it for the first time.
> "Extension" is Linux AMA/AMCS concept and so I used the same name. if you have better name, let me know.

Reply:
amcs_extension?